### PR TITLE
Add homepage based on Figma template

### LINKED ISSUE #41: E: 5. As unauthorized user I want to see main page of app

### PR DESCRIPTION
**Pull request by RMuskovets**
# Checklist:

- [x] Added specs for all new ruby code
- [x] Green rspec run
- [x] Green rubocop
- [x] PR review from teammate

# What?
The home page (aka main page / index / etc.), as close as possible to the Figma template.

## Why?
Because a) it's required on every website AFAICT b) #41.

## How?
A `static` controller with `index` action, invoked on root (`/`) requests. A Slim view for the homepage in `app/views/static/index.html.slim`. And some stylesheets for the page as well.

## Testing?
Did you write tests? No.

## Screenshots (optional)
Screenshot from 2021-05-26 15-26-59
Screenshot from 2021-05-26 15-27-04
Screenshot from 2021-05-26 15-27-06

## Anything Else?
Closes #41.

### CODE REVIEW

File: app/assets/stylesheets/static.scss [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+        p {
+            max-width: 60%;
+            font-size: 1.5em;
+        }
+    }
+
+    .book {
+        border-radius: 20px;
+        backdrop-filter: blur(40px);
```
Comment (montashevych) [inline app/assets/stylesheets/static.scss:45]: i think we need more than 71% supporting devices) but this css feature looks great

**Review by montashevych — APPROVED** [review]
looks fine.

File: app/controllers/static_controller.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class StaticController < ApplicationController
```
Comment (andrjuha01) [inline app/controllers/static_controller.rb:1]: I bet it's not a static page.
We have a search here and places with discounts at the bottom.

File: app/views/static/index.html.slim [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      div.bottom-part
+        span Places to work
+        button.btn.btn-success Explore
+
+div.fw3.d-flex
+  h1.mx-auto Best offers
+
+section.best-offers
+  div.container
+      div.offer
```
Comment (andrjuha01) [inline app/views/static/index.html.slim:56]: make those as links

File: app/controllers/static_controller.rb (hunk 2 above)
Comment (andrjuha01) [reply app/controllers/static_controller.rb:1]: Also add title to the page(it's probably missing)

File: app/assets/stylesheets/static.scss [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+// You can use Sass (SCSS) here: https://sass-lang.com/
+
+html, body {
+    // overflow-x: hidden;
+    background: #262F30 !important;
+}
+
+.main-screen {
+
+    background: url(/packs/media/images/homepage-bg-f0e66c66.png);
```
Comment (montashevych) [inline app/assets/stylesheets/static.scss:12]: why u use path through packs? we can use asset-url("homepage-bg.png"). If in the future will we need change images ?

File: app/views/static/index.html.slim (hunk 3 above)
Comment (RMuskovets) [reply app/views/static/index.html.slim]: done in d2bdc95 if that's what you meant

File: app/assets/stylesheets/static.scss (hunk 4 above)
Comment (RMuskovets) [reply app/assets/stylesheets/static.scss]: fixed in a53f55c

File: app/controllers/static_controller.rb (hunk 2 above)
Comment (RMuskovets) [reply app/controllers/static_controller.rb]: renamed to `HomeController` in 25952d2; added title in 107c599

File: app/helpers/static_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module StaticHelper
```
Comment (andrjuha01) [inline app/helpers/static_helper.rb:1]: remove it

File: spec/helpers/static_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+# the StaticHelper. For example:
+#
+# describe StaticHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe StaticHelper, type: :helper do
```
Comment (andrjuha01) [inline spec/helpers/static_helper_spec.rb:13]: remove this one

File: spec/requests/static_spec.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'Statics', type: :request do
```
Comment (andrjuha01) [inline spec/requests/static_spec.rb:3]: remove this file as well

File: app/controllers/home_controller.rb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+class HomeController < ApplicationController
+  def index; end
```
Comment (andrjuha01) [inline app/controllers/home_controller.rb:2]: add spec for this action(it renders index and returns success)

File: app/assets/stylesheets/static.scss [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+            .information {
+                margin: 36px;
+
+                .title {
+                    font-family: 'Montserrat'; font-weight: 700; font-size: 18pt;
+                    color: var(--bs-light); text-decoration: none;
+                }
+
+                .main-info p, .main-info a
+                    { padding-left: 36px; line-height: 1.2em; display: block; position: relative; }
```
Comment (andrjuha01) [inline app/assets/stylesheets/static.scss:163]: not really readable.
Use multiline css if you have more than one property set

File: app/assets/stylesheets/static.scss [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+                .main-info p:before, .main-info a:before
+                    { font-size: 2em; padding-bottom: -4px; font-family: 'Material Icons'; left: -4px; top: 0.1em; position: absolute; }
```
Comment (andrjuha01) [inline app/assets/stylesheets/static.scss:166]: same here. Review all the file.

File: app/assets/stylesheets/static.scss [hunk 11]
```diff
@@ -0,0 +1,188 @@
+// Place all the styles related to the static controller here.
```
Comment (andrjuha01) [inline app/assets/stylesheets/static.scss:1]: We don't have static controller

File: app/views/home/index.html.slim [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+              p.location 1749 Wheeler Ridge Delaware
+              p.guests 2 Guests
+              p.rooms 1 Room
+            div.price
+              s.old-price $ 4,500
+              div.price-main
+                p.new-price.flex-grow-1 $ 2,500
+                button.btn.btn-success Book Now
+      div.col-lg-4
+        div.offer
```
Comment (andrjuha01) [inline app/views/home/index.html.slim:91]: use links.
If I click on property I should be redirected to it

**Review by andrjuha01 — APPROVED** [review]

**Pull request merged**